### ISSUE TITLE
Document/Permit usage outside a rake task

### ISSUE THREAD

**Issue by jvenezia** [gregnavis/active_record_doctor#23]
Thanks a lot for this gem, really helpful 😄!

We added those to our test suite, ensuring we don't have new database problems in the future.

Here is an example of how we use it : 

```
  describe 'Unindexed Foreign Keys' do
    before { Rails.application.load_tasks }

    let(:allowed) do
      {
        'users' => ['allowed_id'],
        'articles' => %w(other_id allowed_id),
      }
    end

    let(:missing) { ActiveRecordDoctor::Tasks::UnindexedForeignKeys.run.first }

    it do
      expect(allowed.keys).to match_array missing.keys

      allowed.each do |table, attributes|
        expect(attributes).to match_array missing[table]
      end
    end
  end
```

This works great, but I would prefer not having to load rake tasks to be able to do that (`before { Rails.application.load_tasks }`).
Also it would be nice to document this usage in the Readme!

**Comment by gregnavis** [comment]
Thanks for the kind words, @jvenezia. I'm glad you find the gem useful.

I'm actually planning on making the tool usable in standalone form. For example, your case of allowing some exceptions to the unindexed foreign keys rule would be handled by configuring it in `.doctor.yml`. Would that be useful to you? I'm happy to take your use case into account when designing that.

**Comment by jvenezia** [comment]
`.doctor.yml` is definitely a great idea to have a project-wide configuration and exclude false positives.
The tests would then just ensure the output is empty.

**Comment by gregnavis** [comment]
@mtejedorwolox, thank you for the kind words. A new version is in progress so stay tuned!

**Comment by jaredmoody** [comment]
+1 on thanks for this gem :) 

I'd also be interested in using this integrated into a CI that runs all checks each build.

Things I can think of to get here:

Things that seem required:
- Need a way to run all checks at once
- Need a way to configure which checks run and/or to skip false positives
- Proper exit status on fail (may already be complete)

Things that would be nice:
- Progress output
- Have something akin to `.rubocop-todo.yml` (maybe `.doctor-todo.yml`) for getting started where you can keep all checks active to catch new cases but ignore current issues until you can get to them.
- Maybe others?

Let me know how I can help!

**Comment by djezzzl** [comment]
Hey guys, there is another gem (https://github.com/djezzzl/database_consistency) which serves the same purposes but in a different way. We have something in common as some differencies. That gem has configuration file which can disable some checks if needed. 

P.S. I leave this comment just to provide people some alternatives if they want. No plan to hurt `ActiveRecordDoctor` reputation 👍

**Comment by mollerhoj** [comment]
+! Would be great with built in support for adding the checks to specs!

**Comment by gregnavis** [comment]
Big announcement! I've just released v1.9.0.rc1 which supports CI-usage and configurability. Please add the following to `Gemfile`:

```ruby
gem "active_record_doctor", "= 1.9.0.rc1"
```

and run `bundle install`. Detailed installation and usage instructions are in the README. If you stumble upon any issues then please report them here.

**Comment by gregnavis** [comment]
I'm closing the issue as it's addressed in 1.9.0 which I released a few minutes ago. Please consult the updated README for instructions on how to configure the tool.
